daemon/logger/ring.go

```diff
@@ -0,0 +1,219 @@
+package logger
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultRingMaxSize = 1000000 // 1MB
+)
+
+// RingLogger is a ring buffer that implements the Logger interface.
+// This is used when lossy logging is OK.
+type RingLogger struct {
+	buffer *messageRing
+	ctx    context.Context
```

Comment:
Since this is not taking advantage of context features such as hierarchy or value storage, could it just be a regular channel? I've heard contexts inside structures derided as an anti-pattern, though I admit to doing it sometimes.